results.pm

```diff
@@ -439,4 +443,31 @@ sub writelog
 	close($fh);
 }
 
+sub openqa_formatter
+{
+    my ($log_data) = @_;
+    my $json_data = {
+	'environment' => $log_data->{sysinfo},
+	'stats' => $log_data->{tests}->{stats},
+	'results' => []
+    };
+
+    for my $r (@{$log_data->{tests}->{results}}) {
+	my $result = 'unk';
+	$result = 'pass' if ($r->{passed} || $r->{skipped});
+	$result = 'fail' if ($r->{failed} || $r->{broken} || $r->{warnings});
```

Comment:
I wonder what happen if `$result eq 'unk'` not defined due no result. This should not happen (LTP should `TBROK` when no result, but maybe worth to handle that (print warning?)

Reply:
i am not sure if a warning helps here. AFAIK this hasnt happened so far with the tests that we had run for the forked repo. In addition i believe that this should not reach this block of code. Actual it should run only if `my·@result·=·backend::run_cmd($backend,·$cmd);` returns `@log` otherwise `stop_exit`. i am thinking to leave it as it is and create a ticket for the handling in case the `backend::run_cmd` returns `0`(atm seems that runs and then prints the @results and continue without checking if the @result is 0 or @log). WDYT?

Reply:
Sure, it can probably stay as it is. The main word should have @metan-ucw @richiejp @cfconrad.

Reply:
I would say, all this is about this openQA parser https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Parser/Format/LTP.pm#L45 and especially this line where it only know `ok` and `fail`.
Only `pass` will be `ok` everything else is just `fail`, so `unk` as well.

I talked with @b10n1k about that and he mentioned that you had already a discussion about it, where that Parser is coming from and who is already using it. Currently I only see one user of that Parser and thats the `tests/publiccloud/run_ltp.pm`.

So general question is actually, does native LTP also produce json format? If yes, how does it differ from `runltp-ng` json format?

Reply:
@cfconrad Our OpenQA LTP runs use another testrunner that was written before runltp-ng. There is no support for JSON in the upstream LTP at all, the testrunner there is 20 years old and shows it's age. We want to replace that with runltp-ng eventually. Also the format the the OpenQA LTP testrunner produces was handed down to us when we asked for alternative way how to show results in OpenQA and was decided by whoever implemented that functionality.

Reply:
Thx for explanation. That info is useful when we are going to change https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Parser/Format/LTP.pm in the future!

Reply:
@cfconrad FYI our openQA LTP does not use any LTP runner (e.g. [`runltp`](https://github.com/linux-test-project/ltp/blob/master/runltp)). It's run by pure openQA code, which parses LTP runtest file [1] and run everything after first item [2].

[1] https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/LTP/utils.pm#L244
[2] https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/kernel/run_ltp.pm#L322